pkg/ignition/ignition.go

```diff
@@ -102,5 +126,109 @@ func Initialize(c client.Client) error {
 		return errors.Errorf("ignition missing kubelet systemd unit file")
 	}
 	log.Info("extracted", "kubelete systemd unit", RenderedWorker.KubeletUnit)
-	return nil
+	RenderedWorker.BootstrapFiles, err = setupBootstrapFiles(clusterServiceCIDR)
+	return err
+}
+
+// setupBootstrapFiles creates bootstrap kubeconfig, cloud-config, kubelet cert and config files and returns their paths
+func setupBootstrapFiles(clusterServiceCIDR string) ([]string, error) {
+	log := ctrl.Log.WithName("setupBootstrapFiles")
+	log.Info("in setupBootstrapFiles")
+	bootstrapFiles, err := createFilesFromIgnition()
+	if err != nil {
+		return nil, err
+	}
+	log.Info("executed GetFilesFromIgnition", "files", bootstrapFiles)
+
+	clusterDNS, err := cluster.GetDNS(clusterServiceCIDR)
+	if err != nil {
+		return nil, err
+	}
+	initKubeletConfigPath, err := createKubeletConf(clusterDNS)
+	if err != nil {
+		return nil, err
+	}
+	log.Info("created kubelet.conf", "path", initKubeletConfigPath)
+	bootstrapFiles = append(bootstrapFiles, initKubeletConfigPath)
+	log.Info("added bootstrap files", "files to transfer", bootstrapFiles)
+	return bootstrapFiles, nil
+}
+
+func createFilesFromIgnition() ([]string, error) {
+	// For each new file in the ignition file check if is a file we are interested in, if so, decode it
+	// and write the contents to a temporary destination path
+	filesToTranslate := map[string]struct{}{
+		payload.BootstrapKubeconfig: {},
+		payload.KubeletCACert:       {},
+	}
+	results := CloudConfigRegex.FindStringSubmatch(*RenderedWorker.KubeletUnit.Contents)
+	if len(results) == 2 {
+		filesToTranslate[payload.CloudConfigPath] = struct{}{}
+	}
+
+	if _, err := os.Stat(payload.GeneratedDir); errors.Is(err, os.ErrNotExist) {
+		err := os.Mkdir(payload.GeneratedDir, os.ModePerm)
+		if err != nil {
+			return nil, err
+		}
+	}
+	filesFromIgnition := []string{}
+	for _, ignFile := range RenderedWorker.Config.Storage.Files {
+		if _, ok := filesToTranslate[ignFile.Node.Path]; ok {
+			if ignFile.Contents.Source == nil {
+				return nil, errors.Errorf("could not process %s: File is empty", ignFile.Node.Path)
+			}
+			contents, err := dataurl.DecodeString(*ignFile.Contents.Source)
+			if err != nil {
+				return nil, errors.Wrapf(err, "could not decode %s", ignFile.Node.Path)
+			}
+			newContents := contents.Data
+			dest := filepath.Join(payload.GeneratedDir, filepath.Base(ignFile.Node.Path))
+			if err = os.WriteFile(dest, newContents, 0644); err != nil {
+				return nil, fmt.Errorf("could not write to %s: %s", dest, err)
+			}
+			filesFromIgnition = append(filesFromIgnition, dest)
+		}
+	}
+	return filesFromIgnition, nil
+}
+
+// createKubeletConf creates config file for kubelet, with Windows specific configuration
+// Add values in kubelet_config.json files, for additional static fields.
+// Add fields in kubeletConf struct for variable fields
+func createKubeletConf(clusterDNS string) (string, error) {
```

Comment:
Many of the options that were command line are being moved to the conf file. We might hit that when we move to 1.25. I suggest doing that here otherwise we will blocked on moving to 1.25.

Reply:
Sure, this is a TODO.
Putting a list of the deprecated args here for convenience:
```golang
"--logtostderr=false", // deprecated (removed)
"--log-file=" + windows.KubeletLog,  // deprecated (removed), alternative --log-dir also deprecated (removed)
"--register-with-taints=" + ignition.WindowsTaints, // deprecated (moved to config file)
"--container-runtime=remote", // deprecated (removed)
"--resolv-conf=", // deprecated (moved to config file)
```